torch/fx/graph.py

```diff
@@ -131,19 +135,25 @@ def _name(self, target: Target) -> str:
     def python_code(self, root_module: str) -> Tuple[str, str, List[str]]:
         free_vars: List[str] = []
         body: List[str] = []
+
+        def insert_hierarchy_guard(node : Node):
+            if node.module_qualname:
+                body.append(f'with torch.fx.ModuleHierarchyCtxMgr(\'{node.module_qualname}\'):\n    ')
         for node in self.nodes:
             if node.op == 'placeholder':
                 assert isinstance(node.target, str)
                 free_vars.append(node.target)
                 continue
             elif node.op == 'call_method':
                 assert isinstance(node.target, str)
+                insert_hierarchy_guard(node)
```

Comment:
nit: given that `placeholder` case is finishing with `continue` anyway, you can hoist this line out of the `if` conditions and do it only once

Reply:
Don't think that's true? We don't want to emit the hierarchy guard in the case that the node is a placeholder, unless we special case to emit a `pass` statement

Reply:
I meant to write:

```
if node.op == 'placeholder':
  ...
  continue

insert_hierarchy_guard(node)
if node.op == 'call_method':
  ...
elif node.op == 'call_function'
  ...
```